tests/queries/0_stateless/02007_namedtupleitems.sql

```diff
@@ -0,0 +1,33 @@
+SELECT namedTupleItems(tuple(1, 2, 3));
+-- [('1',1),('2',2),('3',3)]
+
+DROP TABLE IF EXISTS test02007;
+CREATE TABLE test02007 (
+       col Tuple(
+           a Tuple(key1 int, key2 int),
+           b Tuple(key1 int, key3 int)
+       )
+) ENGINE=Memory();
+INSERT INTO test02007 VALUES (tuple(tuple(1, 2), tuple(3, 4)));
+INSERT INTO test02007 VALUES (tuple(tuple(5, 6), tuple(7, 8)));
+SELECT namedTupleItems(col) FROM test02007 ORDER BY col;
+-- [('a',(1,2)),('b',(3,4))]
+-- [('a',(5,6)),('b',(7,8))]
+
+DROP TABLE IF EXISTS test02007;
+CREATE TABLE test02007 (
+       col Tuple(CPU double, Memory double, Disk double)
+) ENGINE=Memory();
+INSERT INTO test02007 VALUES (tuple(3.3, 5.5, 6.6));
+SELECT untuple(arrayJoin(namedTupleItems(col))) from test02007;
+-- CPU	3.3
+-- Memory	5.5
+-- Disk	6.6
+
+DROP TABLE IF EXISTS test02007;
+SELECT namedTupleItems(tuple(1, 1.3)); -- { serverError 43; } should it?
```

Comment:
As I understand this tuple contains two different element types UInt8 and Double so exception should be thrown.

Reply:
I was wondering if both values should be casted to the smallest supertype of both types. For example:

```
> SELECT namedTupleItems((toInt8(3), 3))
```

users would maybe expect that both types would be casted to a regular `int`. 

If you don't think this is needed, or if you think this could be done in a subsequent PR, I'm perfectly fine with that.

Reply:
Current implementation is OK. In a lot of places we require user to be more accurate with types and use additional casts if necessary.

Reply:
Sounds good